### PR TITLE
Added peer miniblocks save after start in epoch

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added peer miniblocks save after start in epoch

### CODE REVIEW

**Review by sstanculeanu — APPROVED** [review]

File: epochStart/bootstrap/baseStorageHandler.go [hunk 1]
```diff
@@ -136,3 +138,54 @@ func (bsh *baseStorageHandler) saveMetaHdrForEpochTrigger(metaBlock data.HeaderH
...
+	mbBytes, err := bsh.marshalizer.Marshal(mb)
+	if err != nil {
+		return err
+	}
+
+	return bsh.storageService.Put(dataRetriever.MiniBlockUnit, hash, mbBytes)
+}
+
+func (bsh *baseStorageHandler) saveMiniblocksFromComponents(components *ComponentsNeededForBootstrap) {
+	log.Debug("saving pending miniblocks", "num pending pending miniblocks", len(components.PendingMiniBlocks))
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply epochStart/bootstrap/baseStorageHandler.go]: done

**Review by sstanculeanu — DISMISSED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by sstanculeanu — DISMISSED** [review]

File: update/sync/syncMiniBlocks.go [hunk 2]
```diff
@@ -103,6 +106,14 @@ func (p *pendingMiniBlocks) SyncPendingMiniBlocks(miniBlockHeaders []data.MiniBl
 func (p *pendingMiniBlocks) syncMiniBlocks(listPendingMiniBlocks []data.MiniBlockHeaderHandler, ctx context.Context) error {
 	_ = core.EmptyChannel(p.chReceivedAll)
 
+	// TODO - remove this
```
Comment (AdoAdoAdo) [inline update/sync/syncMiniBlocks.go:109]: is this still required?

File: process/block/baseProcess.go [hunk 3]
```diff
@@ -1267,7 +1267,7 @@ func (bp *baseProcessor) saveBody(body *block.Body, header data.HeaderHandler, h
 		if errNotCritical != nil {
 			log.Warn("saveBody.Put -> MiniBlockUnit", "error", errNotCritical.Error())
 		}
-		log.Trace("saveBody.Put -> MiniBlockUnit", "time", time.Since(startTime))
+		log.Debug("saveBody.Put -> MiniBlockUnit", "time", time.Since(startTime), "hash", miniBlockHash) // TODO revert this to TRACE
```
Comment (sstanculeanu) [inline process/block/baseProcess.go:1270]: missing todo revert

File: storage/pruning/pruningStorer.go [hunk 4]
```diff
@@ -314,9 +315,17 @@ func (ps *PruningStorer) getPersisterToUse() *persisterData {
...
+			returningPath = persisterToUse.path
+		}
+
 		log.Debug("active persister not found",
 			"epoch", ps.epochForPutOperation,
-			"used", persisterToUse.epoch)
+			"used", persisterToUse.epoch,
+			"path", ps.dbPath,
+			"returning persister", returningPath,
+			"stack", string(debug.Stack())) // TODO remove debug.Stack call
```
Comment (sstanculeanu) [inline storage/pruning/pruningStorer.go:328]: missing todo revert

File: update/sync/syncMiniBlocks.go (hunk 2 above)
Comment (iulianpascalau) [reply update/sync/syncMiniBlocks.go:109]: no, removed. Thanks

**Review by sstanculeanu — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**